.buildkite/pipeline.yml

```
@@ -1,73 +1,102 @@
+  matrix:
+    setup:
+      gemfile:
+      - Gemfile
+      - gemfiles/rails_5.gemfile
+      - gemfiles/rails_6_0.gemfile
+
+- label: ":ruby: 2.7 {{matrix.gemfile}} Unit Tests"
+  agents:
+    queue: ruby-2-7-mysql
+    ruby: 2.7
+    elasticsearch: 6.3.1
+  artifact_paths:
+  - spec/reports/**/*.xml
   env:
-    <<: *common_env
-    JUNIT_OUTPUT: spec/reports/rails-4/rspec.xml
+    ELASTICSEARCH_TEST_PORT: '9200'
+    JUNIT_OUTPUT: spec/reports/{{matrix.gemfile}}/rspec.xml
+    BUNDLE_GEMFILE: "{{matrix.gemfile}}"
   commands:
-    - *appraisal_commands
-    - bundle exec appraisal rails-4 rspec
-- label: "Rails 5 Unit Tests"
-  <<: *common_unit_tests
+    - bundle install
+    - bundle exec rspec
+  matrix:
+    setup:
+      gemfile:
+      - Gemfile
+      - gemfiles/rails_5.gemfile
+      - gemfiles/rails_6_0.gemfile
+
+- label: ":ruby: 3.0 {{matrix.gemfile}} Unit Tests"
+  agents:
+    queue: ruby-3-0-mysql
+    ruby: '3.0'
+    elasticsearch: 6.3.1
+  artifact_paths:
+  - spec/reports/**/*.xml
   env:
-    <<: *common_env
-    JUNIT_OUTPUT: spec/reports/rails-5/rspec.xml
+    ELASTICSEARCH_TEST_PORT: '9200'
+    JUNIT_OUTPUT: spec/reports/{{matrix.gemfile}}/rspec.xml
+    BUNDLE_GEMFILE: "{{matrix.gemfile}}"
   commands:
-    - *appraisal_commands
-    - bundle exec appraisal rails-5 rspec
-- label: "Rails 6 Unit Tests"
-  <<: *common_unit_tests
+    - bundle install
+    - bundle exec rspec
+  matrix:
+    setup:
+      gemfile:
+      - Gemfile
+      - gemfiles/rails_6_0.gemfile
+
+- label: ":ruby: 3.1 {{matrix.gemfile}} Unit Tests"
+  agents:
+    queue: ruby-3-1-mysql
+    ruby: 3.1
+    elasticsearch: 6.3.1
+  artifact_paths:
+  - spec/reports/**/*.xml
   env:
-    <<: *common_env
-    JUNIT_OUTPUT: spec/reports/rails-6/rspec.xml
+    ELASTICSEARCH_TEST_PORT: '9200'
+    JUNIT_OUTPUT: spec/reports/{{matrix.gemfile}}/rspec.xml
+    BUNDLE_GEMFILE: "{{matrix.gemfile}}"
   commands:
-    - *appraisal_commands
-    - bundle exec appraisal rails-6 rspec
+    - bundle install
+    - bundle exec rspec
+  matrix:
+    setup:
+      gemfile:
+      - Gemfile
+      - gemfiles/rails_6_0.gemfile
+
 
   # Test Summary Annotate Plugin
 - wait: ~
   continue_on_failure: true
 
 - label: Test Summary Annotate
   timeout_in_minutes: 5
-  <<: *common_agent
+  agents:
+    queue: ruby-2-6
   plugins:
-    - invoca/test-summary#TECH-5520_optional_flag_to_run_native_instead_of_docker:
-        inputs:
-          - label: Current Unit Tests
-            artifact_path: spec/reports/current/rspec.xml
-          - label: Rails 4 Unit Tests
-            artifact_path: spec/reports/rails-4/rspec.xml
-          - label: Rails 5 Unit Tests
-            artifact_path: spec/reports/rails-5/rspec.xml
-          - label: Rails 6 Unit Tests
-            artifact_path: spec/reports/rails-6/rspec.xml
-        run_without_docker: true
+  - bugcrowd/test-summary#master:
+      inputs:
+        - label: Unit Tests
+          artifact_path: spec/reports/**/rspec.xml
```

This is throwing an error in the test summary output: `/buildkite/plugins/github-com-bugcrowd-test-summary-buildkite-plugin-master/lib/test_summary_buildkite_plugin/input.rb:12:in 'create': missing keyword: type (ArgumentError)`
```suggestion
          artifact_path: spec/reports/**/rspec.xml
          type: junit
```